django/__init__.py

```diff
@@ -21,4 +28,18 @@ def setup(set_prefix=True):
         set_script_prefix(
             "/" if settings.FORCE_SCRIPT_NAME is None else settings.FORCE_SCRIPT_NAME
         )
-    apps.populate(settings.INSTALLED_APPS)
+
+    with ExitStack() as stack:
+        for connection in connections.all():
+            stack.enter_context(connection.execute_wrapper(_warn_on_queries))
+        apps.populate(settings.INSTALLED_APPS)
+
+
+def _warn_on_queries(execute, *args):
+    warnings.warn(
+        "Accessing the database during app initialization is discouraged. See"
+        " https://docs.djangoproject.com/en/stable/ref/applications/#troubleshooting"
+        " for some guidance on how to fix this warning.",
```

Comment:
We don't typically link to the docs from warning messages. It's hard to keep the links up-to-date.